WordPressKit/WordPressKit/MediaServiceRemoteXMLRPC.m

```diff
@@ -30,13 +30,40 @@ - (void)getMediaWithID:(NSNumber *)mediaID
 - (void)getMediaLibraryWithSuccess:(void (^)(NSArray *))success
                            failure:(void (^)(NSError *))failure
 {
-    NSArray *parameters = [self defaultXMLRPCArguments];
+    NSMutableArray * mediaResult = [[NSMutableArray alloc] init];
+    [self getMediaLibraryStartOffset:0 media:mediaResult success:success failure:failure];
+}
+
+- (void)getMediaLibraryStartOffset:(NSUInteger)offset
+                             media:(NSMutableArray *)media
```

Comment:
I think I'd prefer to pass an immutable array here – perhaps we could do that, and just construct a new array each time combining the existing + new items to pass on to the next iteration?

Reply:
I was just copying what we did for the REST API, but I'm happy to change that the memory implications shouldn't be too bad.